### PR TITLE
scripts: west_commands: runners: Added optional remote jlink (--ip) flag

### PR DESCRIPTION
**Pull request by sblackstryker**
Adds an optional argument to the West JLink runner, '--ip' which allows flash to be perfomed to a JLink remote server.
The option is useful for in environments where flash to USB is not possible, i.e. inside WSL2.

To use this option JLink Remote Server needs to be running and a device connected.

This was verified by:
1. Running JLink Remote Server in windows environment.
2. Connecting a Nordic developer kit to the JLink Remote Server.
3. Starting a WSL 2 environment.
4. Within WSL2 starting a docker container with the zephyr toolchain
5. Building a zephyr application
6. When built using west command
```
west flash -r jlink --ip 192.168.5.193
```

Result:
JLink Remote Server status changes to show the connection to the developer board
Output from west
```
-- west flash: rebuilding
ninja: no work to do.
-- west flash: using runner jlink
-- runners.jlink: JLink version: 7.82
-- runners.jlink: Command: ['/opt/SEGGER/JLink_V782/JLinkExe', '-ip', '192.168.5.193', '-nogui', '1', '-if', 'swd', '-speed', '4000', '-device', 'nRF52840_xxAA', '-CommanderScript', '/tmp/tmpbpvd7y0zjlink/runner.jlink', '-nogui', '1']
-- runners.jlink: Flashing file: /workarea/build/zephyr/zephyr.hex
```

### CODE REVIEW

**Comment by pdgendt** [comment]
Please review the [commit guidelines](https://docs.zephyrproject.org/3.1.0/contribute/guidelines.html#commit-guidelines), this PR should have a single commit.

**Comment by mbolivar-nordic** [comment]
Yep, please rebase your patch onto the zephyr main branch. There should be no merge commits in your PR.

Here is a short talk which summarizes our PR process: https://www.youtube.com/watch?v=mnVNLM7ynMQ

**Comment by sblackstryker** [comment]
Rebased with main, removed the merge commits from PR

File: scripts/west_commands/runners/jlink.py [hunk 1]
```diff
@@ -97,6 +99,8 @@ def do_add_parser(cls, parser):
                             action=partial(depr_action,
                                            replacement='-i/--dev-id'),
                             help='Deprecated: use -i/--dev-id instead')
+        parser.add_argument('--remote', required=False, dest='ip',
```
Comment (mbolivar-nordic) [inline scripts/west_commands/runners/jlink.py]: Remove `required=False`; that's the default

File: scripts/west_commands/runners/jlink.py [hunk 2]
```diff
@@ -318,6 +323,7 @@ def flash(self, **kwargs):
         self.logger.debug('JLink commander script:\n' +
                           '\n'.join(lines))
 
+
```
Comment (mbolivar-nordic) [inline scripts/west_commands/runners/jlink.py:326]: Please don't make a habit of mixing in unrelated whitespace changes. A commit should do one thing.

Here is a short presentation on our upstreaming process which explains this and more, including the "no merge commits" rule: https://www.youtube.com/watch?v=mnVNLM7ynMQ

File: scripts/west_commands/runners/jlink.py [hunk 3]
```diff
@@ -337,6 +344,8 @@ def flash(self, **kwargs):
                     '-CommanderScript', fname] +
                    (['-nogui', '1'] if self.supports_nogui else []) +
                    self.tool_opt)
+            
+            self.logger.info('Command: {}'.format(cmd))
```
Comment (mbolivar-nordic) [inline scripts/west_commands/runners/jlink.py:348]: Remove this; it's too verbose for the default behavior and it's already printed with `west -v flash` by `self.check_call`

File: scripts/west_commands/runners/jlink.py [hunk 4]
```diff
@@ -99,7 +99,7 @@ def do_add_parser(cls, parser):
                             action=partial(depr_action,
                                            replacement='-i/--dev-id'),
                             help='Deprecated: use -i/--dev-id instead')
-        parser.add_argument('--remote', required=False, dest='ip',
+        parser.add_argument('--ip', required=False, dest='ip',
```
Comment (mbolivar-nordic) [inline scripts/west_commands/runners/jlink.py:102]: This is a fixup commit, which needs to be squashed properly. Please watch the youtube video I linked; it explains this as well.

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
No objections to the feature!

I have a few requests

**Comment by carlescufi** [comment]
@sblackstryker please address the review comments.

**Comment by carlescufi** [comment]
Sorry, closed by mistake!
